File: test/fixtures/typescript/medium/entity_service_500.ts
// TypeScript Fixture (target ~500 lines)
// Generated for benchmark testing

import { Database } from './database';
import { Logger } from './logger';
import { Cache } from './cache';

interface Entity {
    id: number;
    name: string;
    email: string;
    isActive: boolean;
    createdAt: Date;
    updatedAt: Date;
}

interface CreateDto {
    name: string;
    email: string;
}

interface UpdateDto {
    name?: string;
    email?: string;
    isActive?: boolean;
}

type EntityResponse = Promise<Entity | null>;
type EntitiesResponse = Promise<Entity[]>;

class EntityService {
    private db: Database;
    private logger: Logger;
    private cache: Cache<number, Entity>;

    constructor(database: Database, logger: Logger, cache: Cache<number, Entity>) {
        this.db = database;
        this.logger = logger;
        this.cache = cache;
    }


    async operation0(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation0`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation1(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation1`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation2(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation2`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation3(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation3`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation4(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation4`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation5(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation5`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation6(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation6`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation7(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation7`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation8(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation8`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation9(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation9`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation10(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation10`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation11(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation11`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation12(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation12`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation13(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation13`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation14(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation14`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation15(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation15`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation16(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation16`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation17(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation17`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation18(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation18`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation19(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation19`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation20(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation20`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation21(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation21`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation22(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation22`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation23(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation23`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation24(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation24`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation25(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation25`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation26(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation26`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation27(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation27`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation28(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation28`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

}

export { EntityService, Entity, CreateDto, UpdateDto };

File: test/fixtures/rust/medium/entity_service_500.rs
// Rust Fixture (target ~500 lines)
// Generated for benchmark testing

use anyhow::Result;
use chrono::{DateTime, Utc};
use std::sync::Arc;

#[derive(Debug, Clone)]
pub struct Entity {
    pub id: u64,
    pub name: String,
    pub email: String,
    pub is_active: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone)]
pub struct CreateDto {
    pub name: String,
    pub email: String,
}

#[derive(Debug, Clone)]
pub struct UpdateDto {
    pub name: Option<String>,
    pub email: Option<String>,
    pub is_active: Option<bool>,
}

pub struct EntityService {
    db: Arc<Database>,
    logger: Arc<Logger>,
    cache: Arc<Cache<u64, Entity>>,
}

impl EntityService {
    pub fn new(db: Arc<Database>, logger: Arc<Logger>, cache: Arc<Cache<u64, Entity>>) -> Self {
        Self { db, logger, cache }
    }


    pub async fn operation_0(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_1(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_2(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_3(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_4(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_5(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_6(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_7(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_8(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_9(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_10(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_11(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_12(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_13(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_14(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_15(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_16(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_17(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_18(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_19(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_20(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_21(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_22(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_23(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_24(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_25(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_26(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_27(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_28(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_29(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_30(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }

    pub async fn operation_31(&self, id: u64, data: String) -> Result<Option<Entity>> {
        match self.db.query("SELECT * FROM entities WHERE id = ?", &[&id]).await {
            Ok(result) => {
                self.logger.debug(&format!("Fetched {}", id));
                Ok(Some(result.try_into()?))
            },
            Err(e) => {
                self.logger.error(&format!("Operation failed: {}", e));
                Err(e.into())
            }
        }
    }
}
